WordPress/Sniffs/WP/I18nSniff.php

```diff
@@ -253,8 +256,8 @@ protected function check_argument_tokens( PHP_CodeSniffer_File $phpcs_file, $con
 		}
 
 		if ( T_CONSTANT_ENCAPSED_STRING === $tokens[0]['code'] ) {
-			if ( 'domain' === $arg_name && ! empty( $this->text_domain ) && trim( $content, '\'""' ) !== $this->text_domain ) {
-				$phpcs_file->$method( 'Mismatch text domain. Expected \'%s\' but got %s.', $stack_ptr, 'TextDomainMismatch', array( $this->text_domain, $content ) );
+			if ( 'domain' === $arg_name && ! empty( $this->text_domain ) && ! in_array( trim( $content, '\'""' ), $this->text_domain, true ) ) {
+				$phpcs_file->$method( 'Mismatch text domain. Expected \'%s\' but got %s.', $stack_ptr, 'TextDomainMismatch', array( join( "' or '", $this->text_domain ), $content ) );
```

Comment:
Just a habit of mine I guess (and influence from JavaScript). Happy to use `implode` instaed.